mkosi

```diff
@@ -2097,13 +2097,23 @@ SigLevel = Optional TrustedOnly
 Server = {repository_server}
 """)
 
+    @contextlib.contextmanager
+    def stop_gpg_agent() -> None:
+        try:
+            yield
+        finally:
+            # Kill the gpg-agent used by pacman and pacman-key
+            run(['gpgconf', '--homedir', os.path.join(root, 'etc/pacman.d/gnupg'), '--kill', 'all'])
+
     def run_pacman_key(args: List[str]) -> subprocess.CompletedProcess:
         cmdline = [
             "pacman-key",
             "--nocolor",
             "--config", pacman_conf,
         ]
-        return run(cmdline + args, check=True)
+
+        with stop_gpg_agent():
```

Comment:
nitpick, but I think it would make more sense to be `with gpg_agent()`

Reply:
I changed the invocations to use try finally instead. In hindsight it doesn't really seem all that beneficial to use a contextmanager here.